### DISCUSSION TITLE
Access current across a receptor?

### DISCUSSION THREAD

**Discussion by wagdy88** [jonescompneurolab/hnn-core#588] [Q&A] [answered]
Hi everyone, 

I was wondering if there's a way to access the current across pyramidal cell compartments? We are making changes to GABAb receptor that doesn't show up in the model and we wanted to access individual cells to check:
- The current across different compartments of the cell 
- The current across the GABAb receptor of the cell 

Appreciate your help!
Thanks
Mohamed

**Comment by jasmainak** [comment, marked as answer]
In the development version, I believe you can do that using `record_isec`: https://jonescompneurolab.github.io/hnn-core/dev/generated/hnn_core.simulate_dipole.html#hnn_core.simulate_dipole and then access it using `cell_response.isec`

**Reply by ntolley** [reply]
Very timely as this was merged just last month in #502!

To clarify with the development version installed you specify `simulate_dipole(..., record_vsec='all', record_isec='all')` to turn on recordings of voltages and synaptic currents for all compartments in the model.

You'll have to run `pip install -e hnn-core` after cloning the github repository to a local folder to test the development version. Hope that helps!

**Reply by wagdy88** [reply]
Thank you both! I'll try it out and come back for any questions :)

**Comment by wagdy88** [comment]
Hi @ntolley 

I tried the code below with @jasmainak  and it didn't work
```py
import hnn_core
from hnn_core import  jones_2009_model, simulate_dipole
net = jones_2009_model(add_drives_from_params= True)
simulate_dipole(net, tstop = 100, record_vsec= 'all')
vsec = net.cell_response.vsec
print(vsec[0][0].keys())
```
There's no recording from outside the soma

**Reply by ntolley** [reply]
@wagdy88 can you try looking at a different `gid` (the second index of `vsec`)? `gid=0` corresponds to an `L2_basket` cell which only has one compartment in the model (the soma).

You can see the default gid ranges with `print(net.gid_ranges)`
```
{'L2_basket': range(0, 35),
 'L2_pyramidal': range(35, 135),
 'L5_basket': range(135, 170),
 'L5_pyramidal': range(170, 270)}
```

**Reply by wagdy88** [reply]
I just tried it again, and I can see the different keys for each cell. Thanks

**Comment by wagdy88** [comment]
@ntolley and @jasmainak 
I tried to use isec function to record around gabab receptors in soma, but I keep getting this error message:

```py
import hnn_core
from hnn_core import  jones_2009_model, simulate_dipole
net = jones_2009_model(add_drives_from_params= True)
simulate_dipole(net, tstop = 100, record_vsec= 'all')
isec = net.cell_response.isec[0][36]['soma_gabab']
print(isec[0][36].keys())
```
KeyError: 'soma_gabab'

The tutorial mentioned that's how to get it: 
https://jonescompneurolab.github.io/hnn-core/dev/generated/hnn_core.Cell.html

Thoughts?

**Reply by ntolley** [reply]
Can you try `print(net.cell_response.isec[0][36]['soma'].keys()`? You need to specify the name of the section on the neuron you're inspecting, and then the synapse like:

```py
isec = net.cell_response.isec[0][36]['soma']['soma_gabab']
```

Hope that helps!

**Reply by wagdy88** [reply]
I tried your line but I got the error below:
KeyError: 'soma'

**Reply by ntolley** [reply]
Since this is a nested dictionary you can print the keys if you get a key error. Can you try 
```py
print(net.cell_response.isec[0][36].keys())
```
and paste the output?

**Reply by wagdy88** [reply]
Sure, here's the output I got:
dict_keys([])

**Reply by ntolley** [reply]
Oh! I think you forgot to turn on current recordings. You need to enable it with:
```py
dpl = simulate_dipole(..., record_isec='all')
```

**Reply by ntolley** [reply]
To clarify your code above only has `record_vsec='all'` specified so I think you're only recording the voltages right now

**Reply by wagdy88** [reply]
Perfect! Thanks a lot, I just tried it and it works flawlessly :)

**Comment by jasmainak** [comment]
@ntolley Could the tutorial be updated to use `net.gid_ranges` instead of hardcoding a `gid` ?
